coalib/processes/Processing.py

```diff
@@ -373,12 +376,29 @@ def instantiate_processes(section,
                              and the arguments passed to each process which are
                              the same for each object.
     """
-    filename_list = collect_files(
-        glob_list(section.get('files', '')),
-        None,
-        ignored_file_paths=glob_list(section.get('ignore', '')),
-        limit_file_paths=glob_list(section.get('limit_files', '')),
-        section_name=section.name)
+    if section.get('handle_nested', False):
+        filename_list = collect_files(
+            glob_list(section.get('orig_file_name', '')),
+            None,
+            ignored_file_paths=glob_list(section.get('ignore', '')),
+            limit_file_paths=glob_list(section.get('limit_files', '')),
+            section_name=section.name)
+        orig_file_path = glob_list(section.get('orig_file_name', ''))[0]
```

Comment:
Why have we specifically used index 0 here in `orig_file_path`?
Does this mean we will work with only one file?

Reply:
The thing is, I have designed the `nl_section` in such a way that each section contains only `one` file. That means if we have a original nested file `test.py.jj2`, then we create two nl_sections i.e `test.py.jj2_nl_python` and `test.py.jj2_nl_jinja2`.

Similary, if we pass two files using the `--files`, where the nested language in each file is `python` and `jjinja2` then we make `4` sections where each sections is designated for one temporary file.

The reason I decided to stick with `one section` for each each file, because it makes the future work of assembling easier.